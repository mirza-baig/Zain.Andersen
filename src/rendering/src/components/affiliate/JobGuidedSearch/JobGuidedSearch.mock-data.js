const defaultData = {
  rendering: {
    componentName: 'JobGuidedSearch',
    dataSource:
      '/sitecore/content/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Home/ta-guided-search/Data/job-guided-search',
    uid: '39fc84ce-1c6d-4db0-bedd-052969d3c8db',
  },

  fields: {
    jobFacets: [
      {
        id: '9052ce94-b5ec-48c1-adb7-e1efa9867a4e',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-categories/call-center',
        name: 'Call Center',
        displayName: 'Call Center',
        fields: {
          categoryImage: {
            value: {
              src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/AndersenWindows/images/help-me-choose/perfect-match-logo.png?h=23&iar=0&w=200&rev=bf8b79e026ee446eb322723f5fde40e8',
              alt: 'Andersen logo with hamburger menu for Perfect Match tool',
              width: '200',
              height: '23',
            },
          },
          categoryName: {
            value: 'Call Center',
          },
          stepsWithIcons: {
            value: '',
          },
          youTubeVideoId: {
            value: '',
          },
          jobSearchPage: {
            value: {
              href: 'https://www.renewalbyandersen.com/careers/Job-Search#f:jobcategoryname=[Call%20Center]',
              linktype: 'external',
              url: 'https://www.renewalbyandersen.com/careers/Job-Search#f:jobcategoryname=[Call%20Center]',
              anchor: '',
              target: '',
            },
          },
        },
        templateId: 'e514ad99-96c0-4327-b9d9-95d707a37a68',
        templateName: 'Job Category',
      },
      {
        id: '2249fb86-f426-49e5-b1bd-ffac8db2f14b',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-categories/installation',
        name: 'Installation',
        displayName: 'Installation',
        fields: {
          categoryImage: {
            value: {},
          },
          categoryName: {
            value: 'Installation',
          },
          stepsWithIcons: {
            value: '',
          },
          youTubeVideoId: {
            value: '',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: 'e514ad99-96c0-4327-b9d9-95d707a37a68',
        templateName: 'Job Category',
      },
      {
        id: 'dc7a192d-f1f3-4b8a-b95c-9b233de5bff7',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-categories/manufacturing',
        name: 'Manufacturing',
        displayName: 'Manufacturing',
        fields: {
          categoryImage: {
            value: {},
          },
          categoryName: {
            value: 'Manufacturing',
          },
          stepsWithIcons: {
            value: '',
          },
          youTubeVideoId: {
            value: '',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: 'e514ad99-96c0-4327-b9d9-95d707a37a68',
        templateName: 'Job Category',
      },
      {
        id: '1a21c17a-5ee8-4992-aba0-0880af4cd8a4',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-categories/marketing',
        name: 'Marketing',
        displayName: 'Marketing',
        fields: {
          categoryImage: {
            value: {},
          },
          categoryName: {
            value: 'Marketing',
          },
          stepsWithIcons: {
            value: '',
          },
          youTubeVideoId: {
            value: '',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: 'e514ad99-96c0-4327-b9d9-95d707a37a68',
        templateName: 'Job Category',
      },
      {
        id: 'd71ff3ac-ed5c-4b03-a9b2-4b2d320d38a9',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-categories/operations',
        name: 'Operations',
        displayName: 'Operations',
        fields: {
          categoryImage: {
            value: {},
          },
          categoryName: {
            value: 'Operations',
          },
          stepsWithIcons: {
            value: '',
          },
          youTubeVideoId: {
            value: '',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: 'e514ad99-96c0-4327-b9d9-95d707a37a68',
        templateName: 'Job Category',
      },
      {
        id: 'b28c0396-8c42-4910-b0f3-8d5326e93f1c',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-categories/sales',
        name: 'Sales',
        displayName: 'Sales',
        fields: {
          categoryImage: {
            value: {},
          },
          categoryName: {
            value: 'Sales',
          },
          stepsWithIcons: {
            value: '',
          },
          youTubeVideoId: {
            value: '',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: 'e514ad99-96c0-4327-b9d9-95d707a37a68',
        templateName: 'Job Category',
      },
      {
        id: '5202eeb3-fd16-4b81-9cd5-226405232f68',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-types/contract',
        name: 'Contract',
        displayName: 'Contract',
        fields: {
          hoursDescription: {
            value: 'Contract',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: '75650273-680f-48c5-b7b9-db467a25986d',
        templateName: 'Job Hours',
      },
      {
        id: '84dd7c1a-9db4-4fbd-b0c2-6261a44b43ff',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-types/full-time',
        name: 'Full Time',
        displayName: 'Full Time',
        fields: {
          hoursDescription: {
            value: 'Full Time',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: '75650273-680f-48c5-b7b9-db467a25986d',
        templateName: 'Job Hours',
      },
      {
        id: '2af3151e-0817-42ed-94c9-72e3db91d4e2',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-types/internship',
        name: 'Internship',
        displayName: 'Internship',
        fields: {
          hoursDescription: {
            value: 'Internship',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: '75650273-680f-48c5-b7b9-db467a25986d',
        templateName: 'Job Hours',
      },
      {
        id: '7da1f322-43e6-43c1-806d-4f9b70a86324',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-types/part-time',
        name: 'Part Time',
        displayName: 'Part Time',
        fields: {
          hoursDescription: {
            value: 'Part Time',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: '75650273-680f-48c5-b7b9-db467a25986d',
        templateName: 'Job Hours',
      },
      {
        id: 'bc7d6956-8638-440c-911f-3966101bc139',
        url: '/sitecore/content/andersencorporation/renewalbyandersen/global/data-sources/jobs/job-types/remote',
        name: 'Remote',
        displayName: 'Remote',
        fields: {
          hoursDescription: {
            value: 'Remote',
          },
          jobSearchPage: {
            value: {
              href: '/search/',
              text: '',
              anchor: 'f:topics=[Product%20Information]',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{9C45865D-65FC-4AD8-95B3-E4D29409AE3D}',
            },
          },
        },
        templateId: '75650273-680f-48c5-b7b9-db467a25986d',
        templateName: 'Job Hours',
      },
    ],
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: '/sitecore/content/andersencorporation/enterprise-global/enums/headline-levels/heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value: 'Be a part of something big.',
    },
    subheadlineLevel: {
      id: '7f10288a-beb1-4d95-a139-32df40094441',
      url: '/sitecore/content/andersencorporation/enterprise-global/enums/headline-levels/heading-3',
      name: 'Heading 3',
      displayName: 'Heading 3',
      fields: {
        Value: {
          value: 'h3',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    subheadlineText: {
      value: 'Show me jobs related to:',
    },
    body: {
      value:
        "Renewal by Andersen is the full-service replacement window division of Andersen Corporation. North America's largest window and door manufacturer.",
    },
    cta1AriaLabel: {
      value: '',
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: '/sitecore/content/andersencorporation/enterprise-global/enums/cta-icons/arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Link: {
      value: {
        href: '',
      },
    },
    cta1Modal: null,
    cta1ModalLinkText: {
      value: '',
    },
    cta1Style: {
      id: 'dd818850-ec95-4d32-9774-7cc8173e277b',
      url: '/sitecore/content/andersencorporation/enterprise-global/enums/cta-styles/link',
      name: 'Link',
      displayName: 'Link',
      fields: {
        Value: {
          value: 'link',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
