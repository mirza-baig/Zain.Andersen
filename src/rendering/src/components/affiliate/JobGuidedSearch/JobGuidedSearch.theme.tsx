// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const JobGuidedSearchTheme: ThemeFile = {
  aw: {
    classes: {},
  },
  rba: {
    classes: {
      container: 'flex flex-col items-center col-span-12 py-l -mx-m px-m ml:px-xxl bg-light-gray',
      headlineClass:
        'text-theme-text text-center pb-xxs text-sm-xxl md:text-l font-extra-light font-sans md:text-center',
      bodyCopy:
        'col-span-12 md:col-span-10 text-theme-body text-body font-regular mb-l text-center',
      subheadline: 'text-small !font-serif font-bold mb-m ml:mb-s',
      // Buttons - Desktop View
      desktopContainer: 'ml:grid ml:grid-cols-3 ml:gap-s ml:mb-l hidden',
      linkClasses:
        'flex text-black text-button font-serif font-bold hover:underline decoration-[3px] hover:theme-btn-decoration hover:underline-offset-8 disabled:text-gray',
      buttonItem:
        'button-item !text-xs !text-black !bg-light-gray rounded-full !h-[64px] !w-[186px] justify-center border-[2px] border-gray group-hover:border-black hover:cursor-pointer hover:border-black  focus:border-xxs active:border-xxs active:!border-primary focus:!border-primary focus:border-[6px]',
      // Dropdown - Mobile View
      mobileContainer: 'grid grid-row w-[325px] md:w-1/2 ml:hidden',
      dropDownSelectList: `block w-full h-[54px] border border-gray bg-none text-sm-xs !py-xxs !pl-xxs text-theme-text items-center flex mb-m`,
      dropDownSelect: `!cursor-pointer appearance-none border-none text-black text-sm-xs bg-[rgba(0,0,0,0)] w-full rounded-xxxs focus:ring-2 -ml-xxxs focus:!ring-light-gray`,
      dropDownSelectIcon: `relative -ml-ml z-[-1]`,
    },
  },
};
