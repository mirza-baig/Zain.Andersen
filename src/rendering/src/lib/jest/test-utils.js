// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
