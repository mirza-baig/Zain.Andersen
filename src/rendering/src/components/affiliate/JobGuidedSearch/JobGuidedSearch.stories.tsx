// Global
import { Story, Meta } from '@storybook/react';
// Local
import JobGuidedSearch, { JobGuidedSearchProps } from './JobGuidedSearch';
import defaultData from './JobGuidedSearch.mock-data';

export default {
  title: 'Affiliate/JobGuidedSearch',
  component: JobGuidedSearch,
} as Meta;
const Template: Story<JobGuidedSearchProps> = (props) => <JobGuidedSearch {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
