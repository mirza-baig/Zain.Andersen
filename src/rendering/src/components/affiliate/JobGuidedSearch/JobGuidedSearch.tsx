// Global
import { useState } from 'react';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { JobGuidedSearchTheme } from './JobGuidedSearch.theme';
import { Headline } from 'src/helpers/Headline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { Subheadline } from 'src/helpers/Subheadline';
import { SvgIcon } from 'src/helpers/SvgIcon';
import ButtonPrimary, { ButtonPrimaryClasses } from 'src/helpers/Button/buttons/btn--primary';
import { LinkWrapper } from 'src/helpers/LinkWrapper';
import classNames from 'classnames';

export type JobGuidedSearchProps =
  Feature.EnterpriseWeb.RenewalByAndersen.Components.Affiliates.Jobs.JobGuidedSearch.JobGuidedSearch;

type jobFacet = Feature.EnterpriseWeb.RenewalByAndersen.Data.Affiliates.Datasources.JobCategory &
  Feature.EnterpriseWeb.RenewalByAndersen.Data.Affiliates.Datasources.JobHours;
const JobGuidedSearch = (props: JobGuidedSearchProps) => {
  const { themeData } = useTheme(JobGuidedSearchTheme);
  const { themeName } = useTheme();
  const [selectedJobFacet, setSelectedJobFacet] = useState<string>('');
  const [isButtonEnabled, setIsButtonEnabled] = useState(false);

  if (!props.fields) {
    return <></>;
  }

  // handle jobFacets
  const handleJobFacets = (jobFacet: string) => {
    setSelectedJobFacet(jobFacet);
    setIsButtonEnabled(true);
  };

  return (
    <Component variant="full" dataComponent="affiliate/jobguidedsearch" {...props}>
      <div className={themeData.classes.container}>
        <Headline classes={themeData.classes.headlineClass} {...props} />
        <BodyCopy {...props} classes={themeData.classes.bodyCopy} />
        <Subheadline classes={themeData.classes.subheadline} {...props} />
        {/* Desktop View */}
        <div className={themeData.classes.desktopContainer}>
          {/* JobFacets Buttons */}
          {props.fields?.jobFacets?.map((_item: jobFacet, i: number) => {
            return (
              <div key={i}>
                <ButtonPrimary
                  field={{
                    value: {
                      href: _item?.fields?.jobSearchPage?.value?.href,
                      anchor: _item?.fields?.jobSearchPage?.value?.anchor,
                      text:
                        _item?.fields?.categoryName?.value ||
                        _item?.fields?.hoursDescription?.value,
                      target: _item?.fields?.jobSearchPage?.value?.target || '_self',
                    },
                  }}
                  variant={undefined}
                  classes={themeData.classes.buttonItem}
                />
              </div>
            );
          })}
        </div>
        {/* Mobile View */}
        <div className={themeData.classes.mobileContainer}>
          {/* JobFacets Dropdown */}
          <div className={themeData.classes.dropDownSelectList}>
            <select
              className={themeData.classes.dropDownSelect}
              onChange={(e) => handleJobFacets(e.target.value)}
              value={selectedJobFacet}
            >
              <option value="" defaultValue="" disabled>
                Select
              </option>
              {props.fields?.jobFacets.map((option: jobFacet, index: number) => (
                <option value={index} key={index}>
                  {option.fields?.categoryName?.value || option.fields?.hoursDescription?.value}
                </option>
              ))}
            </select>
            <SvgIcon
              icon={'dropdown-arrow'}
              size="20"
              className={themeData.classes.dropDownSelectIcon}
            ></SvgIcon>
          </div>
          <ButtonPrimary
            variant={undefined}
            classes={classNames(
              ButtonPrimaryClasses(themeName).btnClass,
              isButtonEnabled === false &&
                '!bg-gray !text-dark-gray pointer-events-none [&_span]:text-dark-gray',
              '!w-full !justify-center mb-l'
            )}
            icon={{
              id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
              url: '/sitecore/content/andersencorporation/enterprise-global/enums/cta-icons/arrow',
              name: 'Arrow',
              displayName: 'Arrow',
              fields: {
                Value: {
                  value: 'arrow',
                },
              },
              templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
              templateName: 'Enum',
            }}
            field={{
              value: {
                href:
                  props.fields?.jobFacets?.[selectedJobFacet]?.fields?.jobSearchPage?.value?.href ||
                  '',
                text: 'Search',
                linktype:
                  props.fields?.jobFacets?.[selectedJobFacet]?.fields?.jobSearchPage?.value
                    ?.external,
                url: props.fields?.jobFacets?.[selectedJobFacet]?.fields?.jobSearchPage?.value?.url,
                anchor:
                  props.fields?.jobFacets?.[selectedJobFacet]?.fields?.jobSearchPage?.value?.anchor,
                target:
                  props.fields?.jobFacets?.[selectedJobFacet]?.fields?.jobSearchPage?.value
                    ?.target || '_self',
              },
            }}
          />
        </div>
        {/* CTA Link */}
        <LinkWrapper
          className={themeData.classes.linkClasses}
          field={{
            href: props.fields?.cta1Link?.value?.href || '',
            title: props.fields?.cta1Link?.value?.text,
            text: props.fields?.cta1Link?.value?.text,
            target: props.fields?.cta1Link?.value?.target || '_self',
          }}
          ariaLabel={{ value: props.fields?.cta1Link?.value?.text || 'arrow icon' }}
        >
          <SvgIcon className="ml-[10px] text-primary" icon="arrow" size="sm" />
        </LinkWrapper>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<JobGuidedSearchProps>(JobGuidedSearch);
