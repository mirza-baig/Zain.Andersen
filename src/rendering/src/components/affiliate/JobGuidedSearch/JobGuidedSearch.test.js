// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import JobGuidedSearch from './JobGuidedSearch';
import defaultData from './JobGuidedSearch.mock-data';

it('renders correctly', async () => {
  const component = snapshot(JobGuidedSearch, { componentProps: defaultData });
  await hasDataComponent(component, 'affiliate/jobguidedsearch');
});
