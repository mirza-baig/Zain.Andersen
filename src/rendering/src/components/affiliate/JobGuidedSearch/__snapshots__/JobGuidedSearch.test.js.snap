// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="affiliate/jobguidedsearch"
  id="id88722135"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div>
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-m md:text-m font-bold mb-s"
        >
          <h2>
            Be a part of something big.
          </h2>
        </div>
        <div
          class="text-theme-body text-body mb-s"
        >
          <div
            class="body-copy"
          >
            <div>
              Renewal by Andersen is the full-service replacement window division of Andersen Corporation. North America's largest window and door manufacturer.
            </div>
          </div>
        </div>
        <div
          class="font-sans"
        >
          <h3>
            Show me jobs related to:
          </h3>
        </div>
        <div>
          <div>
            <a
              aria-label="Call Center"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="https://www.renewalbyandersen.com/careers/Job-Search#f:jobcategoryname=[Call%20Center]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Call Center
            </a>
          </div>
          <div>
            <a
              aria-label="Installation"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Installation
            </a>
          </div>
          <div>
            <a
              aria-label="Manufacturing"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Manufacturing
            </a>
          </div>
          <div>
            <a
              aria-label="Marketing"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Marketing
            </a>
          </div>
          <div>
            <a
              aria-label="Operations"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Operations
            </a>
          </div>
          <div>
            <a
              aria-label="Sales"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Sales
            </a>
          </div>
          <div>
            <a
              aria-label="Contract"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Contract
            </a>
          </div>
          <div>
            <a
              aria-label="Full Time"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Full Time
            </a>
          </div>
          <div>
            <a
              aria-label="Internship"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Internship
            </a>
          </div>
          <div>
            <a
              aria-label="Part Time"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Part Time
            </a>
          </div>
          <div>
            <a
              aria-label="Remote"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/search/#f:topics=[Product%20Information]"
              role="link"
              tabindex="0"
              target="_self"
            >
              Remote
            </a>
          </div>
        </div>
        <div>
          <div>
            <select>
              <option
                disabled=""
                value=""
              >
                Select
              </option>
              <option
                value="0"
              >
                Call Center
              </option>
              <option
                value="1"
              >
                Installation
              </option>
              <option
                value="2"
              >
                Manufacturing
              </option>
              <option
                value="3"
              >
                Marketing
              </option>
              <option
                value="4"
              >
                Operations
              </option>
              <option
                value="5"
              >
                Sales
              </option>
              <option
                value="6"
              >
                Contract
              </option>
              <option
                value="7"
              >
                Full Time
              </option>
              <option
                value="8"
              >
                Internship
              </option>
              <option
                value="9"
              >
                Part Time
              </option>
              <option
                value="10"
              >
                Remote
              </option>
            </select>
            <span />
          </div>
          <a
            aria-label="Search"
            class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed !bg-gray !text-dark-gray pointer-events-none [&_span]:text-dark-gray !w-full !justify-center mb-l"
            data-gtm-click=""
            data-gtm-dl-event="cta_click"
            href=""
            role="link"
            tabindex="0"
            target="_self"
          >
            Search
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
